doc/source/dev/contributor/debugging_linalg_issues.rst

```diff
@@ -0,0 +1,440 @@
+    $ cd ../..
+
+Build FlexiBLAS::
+
+    $ mkdir build && cd build
+    $ # Note: this will also pick up the libraries in your system/env libdir
+    $ cmake .. -DEXTRA="OpenBLAS" -DLAPACK_API_VERSION=3.9.0 \
+        -DOpenBLAS_LIBRARY=$PWD/../../built-libs/lib/libopenblas.so \
+        -DCMAKE_INSTALL_PREFIX=$PWD/../../built-libs
+    $ cmake --build . -j
+    $ cmake --install . --prefix $PWD/../../built-libs
+    $ cd ../..
+
+We're now ready to build SciPy against FlexiBLAS::
+
+    $ cd scipy
+    $ python dev.py build -C-Dblas=flexiblas -C-Dlapack=flexiblas
+    ...
+    Run-time dependency flexiblas found: YES 3.4.2
+
+Now we can run the tests. Note that the ``NETLIB`` option is built without
+having to specify it; it's the default in FlexiBLAS and sources are included in
+its repository::
+
+    $ FLEXIBLAS=OpenBLAS python dev.py test -s linalg
+    $ FLEXIBLAS=NETLIB python dev.py test -s linalg
+
+Other libraries available on the system can be inspected with::
+
+    $ ./flexiblas-setup/built-libs/bin/flexiblas list
+
+.. note::
+
+    Using local builds of reference BLAS/LAPACK or BLIS is more difficult,
+    because FlexiBLAS requires a single shared library which contains all
+    needed symbols. It `may be feasible
+    <https://github.com/mpimd-csc/flexiblas#setup-with-precompiled-reference-blas-and-lapack>`__
+    to use a separate ``libblas`` and ``liblapack`` as the "system library",
+    but this has proven to be more fragile and difficult to build (so this is
+    YMMV). In case you do want to try:
+
+    Build reference BLAS and LAPACK:
+
+        $ git clone https://github.com/Reference-LAPACK/lapack.git
+        $ cd lapack
+        $ mkdir build && cd build
+        $ cmake -DCBLAS=ON -DBUILD_SHARED_LIBS=OFF ..
+        $ cmake --build . -j
+        $ cmake --install . --prefix $PWD/../../built-libs
+
+    Then add the following two lines to the ``cmake ..`` configure command for
+    FlexiBLAS::
+
+        -DSYS_BLAS_LIBRARY=$PWD/../../built-libs/lib/libblas.a \
+        -DSYS_LAPACK_LIBRARY=$PWD/../../built-libs/lib/liblapack.a \
+
+
+Creating reproducers in C or Fortran
+------------------------------------
+
+Our experience tells us that a large majority of bugs are inside SciPy rather
+than in OpenBLAS or another BLAS library. If the testing with different BLAS
+libraries tells us though that the problem is specific to a single BLAS library
+(maybe even a single version of that library with a regression), the next step
+is to produce a reproducer in C or Fortran; doing so is necessary for reporting
+the bug upstream, and makes it much easier for the BLAS library developers to
+address the problem.
+
+To get from a Python reproducer which uses a ``scipy`` function with NumPy
+arrays as input to a C/Fortran reproducer, it is necessary to find the code
+path taken in SciPy and determine which exact BLAS or LAPACK function is
+called, and with what inputs. This can then be reproduced in C/Fortran
```

Comment:
to find out what SciPy is doing undr the good, you'd typically need to look into the f2py pyf.src wrapper in the scipy/linalg directory. It might also be helpful to look into the _flapackmodule.c which is generated by f2py. 

Argument lists of LAPACK functions can be looked up on either netlib or Reference-Lapack github repository (would be helpful to add links here, I think)